external/loaders/loaders/mappers/_fine_res.py

```diff
@@ -0,0 +1,236 @@
+from datetime import timedelta
+from enum import Enum
+from typing_extensions import Protocol
+from typing import Optional, Sequence
+import xarray as xr
+import numpy as np
+import fsspec
+
+from vcm.fv3.metadata import gfdl_to_standard
+from loaders._config import mapper_functions
+from loaders.mappers._base import GeoMapper
+from loaders.mappers._xarray import XarrayMapper
+from loaders.mappers._fine_res_budget import compute_fine_res_sources, FineResBudget
+
+class MLTendencies(Protocol):
+    dQ1: xr.DataArray
+    dQ2: xr.DataArray
+
+
+def open_zarr(url, consolidated=False):
+    mapper = fsspec.get_mapper(url)
+    return xr.open_zarr(mapper, consolidated=consolidated)
+
+
+def open_zarr_maybe_consolidated(url):
+    try:
+        return open_zarr(url, consolidated=True)
+    except KeyError:
+        return open_zarr(url, consolidated=False)
+
+
+def standardize_coords(
+    ds: xr.Dataset, time_shift=-timedelta(minutes=7, seconds=30)
+) -> xr.Dataset:
+    ds_shifted = ds.assign(time=ds.time + time_shift)
+    return gfdl_to_standard(ds_shifted).drop("tile")
+
+
+def _open_fine_resolution_dataset(
+    fine_url: str, additional_dataset_urls: Optional[Sequence[str]]
+) -> FineResBudget:
+
+    fine = open_zarr_maybe_consolidated(fine_url)
+    fine_shifted = standardize_coords(fine)
+
+    if additional_dataset_urls is not None:
+        additional_datasets = []
+        for url in additional_dataset_urls:
+            additional_datasets.append(open_zarr_maybe_consolidated(url))
+        merged = xr.merge([fine_shifted, *additional_datasets], join="inner")
+        if "latitude" in merged:
+            merged["latitude"] = merged.latitude.isel(time=0)
+        if "longitude" in merged:
+            merged["longitude"] = merged.longitude.isel(time=0)
+    else:
+        merged = fine_shifted
```

Comment:
In the case where no additional datasets are provided, the previous renaming of the fine-res dataset input features (e.g., `T` -> `air_temperature`) got dropped. I could put it back in but it seems like of klunky to have here and not having it doesn't actually prevent training a model with those features.

Reply:
Putting them back in to enforce that temp and sphum come from the fine-res data, as discussed